### ISSUE TITLE
Introduce `Http` effect

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#9035]
Introduce an effect like:

```
type alias HttpResponse = {status = Int32, statusText = String, url = String, body = String}

eff Http {
    pub def get(url: String, headers: Map[String, String]): HttpResponse
}
```

We will probably need to iterate on the API.

**Comment by magnus-madsen** [comment]
I think the agreement was that HttpResponse should not be a record, but rather an enum and then we can have accessor methods.

**Comment by magnus-madsen** [comment]
Not closed yet. Need the handler and some iterations :)

**Comment by sockmaster27** [comment]
Is the handler part of the high priority?

**Comment by magnus-madsen** [comment]
Yes. Otherwise you cannot use it. But javas new http client should be easy.

**Comment by sockmaster27** [comment]
@magnus-madsen 
I don't really see how we can do this without using java generics.
The `send` operation looks [like this](https://download.java.net/java/early_access/loom/docs/api/java.net.http/java/net/http/HttpClient.html#send(java.net.http.HttpRequest,java.net.http.HttpResponse.BodyHandler)).
If we want a map of headers we need to get it via one of [these methods](https://download.java.net/java/early_access/loom/docs/api/java.net.http/java/net/http/HttpHeaders.html#method-summary). The only possibility there is parsing the string ourselves, but that doesn't seem great.

**Comment by magnus-madsen** [comment]
If the new HttpClient absolutely does not work, then we can use openConnection on URL class.

**Comment by magnus-madsen** [comment]
@sockmaster27 Will you brainstorm some ideas for how to extend Http to be more useful?

**Comment by magnus-madsen** [comment]
And write them up before starting on programming

**Comment by sockmaster27** [comment]
Add operations for each request method:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

I think you were right about splitting them into different operations, to enforce which of them take bodies and which of them don't.

**Comment by sockmaster27** [comment]
Just the methods. Did you have anything else in mind?

**Comment by magnus-madsen** [comment]
> Just the methods. Did you have anything else in mind?

What other functionality is offered by HttpClient or Node's http: https://nodejs.org/api/http.html 

Like what functionality would one expect?

**Comment by sockmaster27** [comment]
> What other functionality is offered by HttpClient or Node's http: https://nodejs.org/api/http.html

It looks like most of what's mentioned on Node is related to hosting a server. Do we want to expand the scope to include that too?
Other than that:
- TLS configurations
- Request timeouts
- Configuration of custom cookie management engines

**Comment by magnus-madsen** [comment]
I think timeouts would be useful, it it can be done very simply? The rest not so much.

**Comment by magnus-madsen** [comment]
I think the next big thing is to have HttpWithResult following the pattern from files.

**Comment by magnus-madsen** [comment]
Oh and the question of how to deal with binary files and big files? What's the story here? This probably more important than timeouts.
